linera-core/src/client/mod.rs

```diff
@@ -1203,143 +1159,144 @@ where
             .known_committees()
             .await
             .map_err(|_| NodeError::InvalidChainInfoResponse)?;
+
         // Retrieve newly received certificates from this validator.
         let query = ChainInfoQuery::new(chain_id).with_received_log_excluding_first_n(tracker);
         let info = remote_node.handle_chain_info_query(query).await?;
-        let mut tracker = MultichainTracker::new(tracker);
         let mut other_sender_chains = Vec::new();
-        let remote_log = info.requested_received_log.clone();
-        let remote_node_chains_view = info.requested_received_log.into_iter().fold(
-            BTreeMap::<ChainId, Vec<BlockHeight>>::new(),
+        let remote_log = info.requested_received_log;
+        let remote_node_chains_view = remote_log.iter().fold(
```

Comment:
`remote_chain_highest`? It now holds a single block height, as opposed to the vector previously.